cpp/src/reductions/reduction_dispatcher_multistep.cuh

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_REDUCTION_DISPATCHER_MULTISTEP_CUH
+#define CUDF_REDUCTION_DISPATCHER_MULTISTEP_CUH
+
+#include "reduction_functions.cuh"
+
+namespace cudf {
+namespace reductions {
+
+// Reduction for mean, var, std
+// It requires extra step after single step reduction call
+template<typename T_in, typename T_out, typename Op, bool has_nulls>
+void ReduceMultiStepOp(const gdf_column *input,
```

Comment:
How about instead of "multi step", we refer to these kinds of reductions as "compound". 
```suggestion
void compound_reduction(const gdf_column *input,
```